### ISSUE TITLE
Fixed Chatroom Name, Warning, Banning Reason and Promote Message Exploits

### ISSUE THREAD

**Issue by Antonhs30000** [jacob-grahn/platform-racing-2-server#28]
@jacob-grahn File: pr2_moderation.php

[Urgent] People can be keylogged using the Chatroom name exploit. Example: `<img src="http://grabifylink" />`, That will grab the IP from the people in the chatroom they're in. 

![image](https://user-images.githubusercontent.com/35195062/34834345-7be22d6c-f6fa-11e7-8472-4ab5e52e5e4f.png)

I fixed it using htmlspecialchars. ^^

```PHP
         htmlspecialchars($welcome_message = 'systemChat`Welcome to chat room '.$this->chat_room_name.'! ');
		if(count($this->player_array) <= 1){
			$welcome_message .= 'You\'re the only person here!';
		}
```
__________________________________________________________________________________________________

[Urgent] Private Server owners have Temporary Moderator abilities. Typing `/view <img src="http://grabifylink" />` and giving a warning will grab the IP Address from the users in the Private Server chatroom. Also typing `/view <a` (works will any character after the "<") will break the chatroom and players won't be able to type anything.

![image](https://user-images.githubusercontent.com/35195062/34835330-2bff4782-f6fd-11e7-9011-fb02da7ae424.png)

__________________________

![image](https://user-images.githubusercontent.com/35195062/34834878-e6660068-f6fb-11e7-91f0-30d26877da1f.png)

Code:

```PHP
		    if(isset($player->chat_room)){
			htmlspecialchars($player->chat_room->send_chat('systemChat`'.$player->name
			.' has given '.$name.' '.$num.' '.$w_str.'. '
			.'They have been silenced from the chat for '.$time.' seconds. Warning evading will result to another warning.', $player->user_id));
		}
```
__________________________________________________________________________________________________

[Not Urgent] HTML in Banning reason message.

Jiggmin has banned bls1999 for 1 year. Reason: **Banned for being a better Admin than me. This ban has been recorded at http://pr2hub.com/bans.**

Code: 			
```PHP
			if($reason != ''){
		htmlspecialchars($disp_reason = 'Reason: '.$reason);
	}
```

_________________________________

[Not urgent] HTML in promote message.

bls1999 has promoted **test to a Temporary Moderator, may they reign in days of peace and prosperity. Make sure you read the moderator guidelines at https://jiggmin2.com/forums/showthread.php?tid=12**

____________________________________

These exploits can be fixed with `strip_tags` as well.

**Comment by bls1999** [comment]
I can fix this as part of pull #24.  I agree that the image exploit is an issue, but the b, u, i, and li html tags shouldn't be banned.  Also, the warning text is fine as it is.

**Comment by Antonhs30000** [comment]
Well, chat will break if you type <a xD

**Comment by bls1999** [comment]
Fixed in #24

**Comment by Antonhs30000** [comment]
Nice!
